js/buffer.ts

```diff
@@ -226,6 +226,42 @@ export class Buffer implements Reader, Writer {
       }
     }
   }
+
+  async readFromMultiple(...r: Reader[]): Promise<number> {
```

Comment:
@ry PTAL at this method. It's not doing the right thing at the moment, my guess is it should read concurrently from stderr and stdin. Which leads me to believe I need to create coroutines for each reader and synchronize them with some kind of deferred object. Is there an easier way 😅 ?

Reply:
@bartlomieju Actually, I think (maybe) there's a lower-level way to do this that will do this automatically. So this whole bit where we specify "piped" or "null" for the stdio streams, maps into a Rust call [like this](https://doc.rust-lang.org/std/process/struct.Command.html#method.stdout)
```rust
Command::new("ls")
        .stderr(Stdio::null())
        .spawn()
        .expect("ls command failed to start");
``` 
What I think should be possible, is to specify something like this:
```rust
Command::new("ls")
        .stderr(std::io::Stdout)
        .spawn()
        .expect("ls command failed to start");
```
This would be done somehow in the `subprocess_stdio_map()` function. See
https://github.com/denoland/deno/blob/034e2cc02829c9244b32232074c7a48af827a2fb/src/ops.rs#L1675-L1677

So - this seems a bit more complex that I originally thought. If you back up this branch to how it was before, but leave the `stderrOutput()` rename, I will land it without the `combinedOutput`. But let's add a TODO to add the combinedOutput.

Reply:
I don't understand fully... Do you want to redirect subprocess stderr to its stdout during creation or more generally be able to provide handle to some file?

Eg.
```ts
const file = await Deno.open("/foo/bar.txt");

Deno.run({
  args: [
    "python",
    "-c",
    "import sys; sys.stderr.write('error'); sys.stdout.write('output')"
  ],
  stdout: file,
  stderr: file,
});
```

Reply:
CC @ry

Reply:
> Do you want to redirect subprocess stderr to its stdout during creation or more generally be able to provide handle to some file?

Redirect stderr to stdout during creation. The more general option would be great, but I think unnecessary for now.

Reply:
Sure I can add that option but that would mean adding `ProcessStdio` option that is only valid for stderr. 

Maybe add another type?
```ts
type ProcessStdout = "inherit" | "piped" | "stdin" | "null" ;
```

Reply:
Hmm. Ideally we would handle the general case of redirecting stdio to an arbitrary file descriptor...

Reply:
That's a good idea, but I see one problem. When you open a file with Deno you get `rid` which is a number. We'd have to differentiate between `rid` and standard file descriptors. I'd suggest to revert changes regarding `combinedOutput` from this PR so we can merge `errorOutput`. 

I'll open separate issue for redirecting subprocess stdio so we can settle on some solution and follow up with another PR.